opus/application/static_media/js/widgets.js

```diff
@@ -394,198 +394,178 @@ var o_widgets = {
             opus.widget_elements_drawn.unshift(slug);
 
         }
-        $.ajax({ url: "/opus/__forms/widget/" + slug + '.html?' + o_hash.getHash(),
-                 success: function(widget_str) {
+        $.ajax({
+            url: "/opus/__forms/widget/" + slug + '.html?' + o_hash.getHash(),
+            success: function(widget_str) {
                 $("#widget__"+slug).html(widget_str);
-            }}).done(function() {
-
-             // if we are drawing a range widget we need to check if the qtype dropdown is
-             // already defined by the url:
-             if (slug.match(/.*(1|2)/)) {
-               // this is a range widget
-                 let id = slug.match(/(.*)[1|2]/)[1];
-
-                // is the qtype constrained in the url?
-                 if ($.inArray('qtype-' + id,opus.extras) > -1 && opus.extras['qtype-'+id]) {
-                     // this widgets dropdown is defined in the url, update the html select dropdown to match
-                     $('#' + widget + ' select').attr("value",opus.extras['qtype-'+id]);
-                 }
-
-                 // add the helper icon to range widgets.
-                // add the helper text for range widgets
-                if ($('.' + widget).hasClass('range-widget') && $('.' + widget).find('select').length) {
-                    // this is a range widget and also the any/all/only dropdown is included
-                    let help_icon = '<a href = "#" data-toggle="popover" data-placement="left">\
-                                    <i class="fa fa-info-circle"></i></a>';
-
-                    $('.' + widget + ' .widget-main>ul>ul').append('<li class = "range-qtype-helper">' + help_icon + '</li>');
+            }
+        }).done(function() {
+            // If there is no specified qtype in the url, we want default qtype to be in the url
+            // This will also put qtype in the url when a widget with qtype is open.
+            // Need to wait until api return to determine if the widget has qtype selections
+            let hash = o_hash.getHashArray();
+            let qtype = "qtype-" + slug;
+            if ($(`#widget__${slug} select[name="${qtype}"]`).length !== 0 && !hash[qtype]) {
+                let defaultOption = $(`#widget__${slug} select[name="${qtype}"]`).first("option").val();
```

Comment:
it's not clear to me why you are getting .first("option").

Reply:
It's because when we open a widget with qtype, the value of the first option tag (under select tag) is the default displayed value for qtype.